File: backend/app/schemas/staff_attendance.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class AttendanceResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    staff_id: int
    clock_in: datetime
    clock_out: datetime | None = None


class AttendanceStatusResponse(BaseModel):
    is_clocked_in: bool
    attendance: AttendanceResponse | None = None
    worked_seconds: int = 0

File: frontend/src/api/StaffAttendance.ts
import { api } from "./client";

export interface AttendanceRecord {
  id: number;
  staff_id: number;
  clock_in: string;
  clock_out: string | null;
}

export interface AttendanceStatus {
  is_clocked_in: boolean;
  attendance: AttendanceRecord | null;
  worked_seconds: number;
}

export async function clockIn(): Promise<AttendanceRecord> {
  const response = await api.post<AttendanceRecord>(
    "/staff/attendance/clock-in",
  );

  return response.data;
}

export async function clockOut(): Promise<AttendanceRecord> {
  const response = await api.post<AttendanceRecord>(
    "/staff/attendance/clock-out",
  );

  return response.data;
}

export async function getMyAttendanceStatus(): Promise<AttendanceStatus> {
  const response = await api.get<AttendanceStatus>(
    "/staff/attendance/me",
  );

  return response.data;
}

export async function getMyAttendanceHistory(): Promise<AttendanceRecord[]> {
  const response = await api.get<AttendanceRecord[]>(
    "/staff/attendance/history",
  );

  return response.data;
}